packages/dashboard-frontend/src/services/workspace-client/devworkspace/devWorkspaceClient.ts

```diff
@@ -400,7 +400,49 @@ export class DevWorkspaceClient extends WorkspaceClient {
     await DwApi.deleteWorkspace(namespace, name);
   }
 
-  async changeWorkspaceStatus(namespace: string, name: string, started: boolean): Promise<devfileApi.DevWorkspace> {
+  getDebugMode(workspace: devfileApi.DevWorkspace): boolean {
+    if (!workspace.metadata.annotations) {
+      return false;
+    }
+    return workspace.metadata.annotations[DEVWORKSPACE_DEBUG_START_ANNOTATION] === 'true';
+  }
+
+  async updateDebugMode(namespace: string, name: string, debugMode: boolean): Promise<devfileApi.DevWorkspace> {
+    const patch: IPatch[] = [];
+    const workspace = await this.getWorkspaceByName(namespace, name);
+    const currentDebugMode = this.getDebugMode(workspace);
+
+    if (currentDebugMode === debugMode) {
+      return workspace;
+    }
+
+    if (!debugMode) {
+      patch.push({
+        op: 'remove',
+        path: '/metadata/annotations',
+        value: DEVWORKSPACE_DEBUG_START_ANNOTATION
+      });
+    } else {
+      if (workspace.metadata.annotations && workspace.metadata.annotations[DEVWORKSPACE_DEBUG_START_ANNOTATION]) {
+        patch.push({
+          op: 'remove',
+          path: '/metadata/annotations',
+          value: DEVWORKSPACE_DEBUG_START_ANNOTATION
+        });
+      }
+      patch.push({
+        op: 'add',
+        path: '/metadata/annotations',
+        value: {
+          [DEVWORKSPACE_DEBUG_START_ANNOTATION]: 'true'
+        }
+      });
```

Comment:
Why don't you use `op: 'replace'` if the value is defined?

Reply:
Ok. I have changed it.